src/coreclr/ilasm/asmman.cpp

```diff
@@ -437,7 +437,7 @@ void    AsmMan::EndAssembly()
                     // Read the file into the buffer.
                     size_t dwBytesRead;
                     
-                    if ((dwBytesRead = fread(m_sStrongName.m_pbPublicKey, 1, m_sStrongName.m_cbPublicKey, fp)) <= m_sStrongName.m_cbPublicKey) {
+                    if ((dwBytesRead = fread(m_sStrongName.m_pbPublicKey, 1, m_sStrongName.m_cbPublicKey, fp)) < m_sStrongName.m_cbPublicKey) {
                         HRESULT hr = HRESULTFromErrno();
```

Comment:
Is `HRESULTFromErrno` correct here? 

The docs for https://en.cppreference.com/w/c/io/fread do not say anything about errno.